### PR TITLE
Promoted dist_tag method to a DSL method

### PR DESCRIPTION
**Pull request by pradeepchhetri**
This makes `dist_tag` configurable. You can disable `dist_tag` in project definition like this:

Disabling `dist_tag` is useful if you are building one rpm for all RPM based distribution and for all their releases.

```
package :rpm do
    dist_tag false
end
```

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @pradeepchhetri

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

File: lib/omnibus/packagers/rpm.rb [hunk 1]
```diff
@@ -210,7 +239,11 @@ def category(val = NULL)
     # @return [String]
     #
     def package_name
-      "#{safe_base_package_name}-#{safe_version}-#{safe_build_iteration}#{dist_tag}.#{safe_architecture}.rpm"
+      if dist_tag
+        "#{safe_base_package_name}-#{safe_version}-#{safe_build_iteration}#{dist_tag}.#{safe_architecture}.rpm"
+      else
+        "#{safe_base_package_name}-#{safe_version}-#{safe_build_iteration}.#{safe_architecture}.rpm"
+      end
     end
```
Comment (schisamo) [inline lib/omnibus/packagers/rpm.rb]: We'll need some unit test coverage form this branching logic.

**Comment by schisamo** [comment]
You also need to update the default SPEC template to check for a false `dist_tag` value:
https://github.com/chef/omnibus/blob/49f6f2197fc86896aacf6fdc542e964e5547d400/resources/rpm/spec.erb#L20

Something like this should work:
```
Release: <%= iteration %><%= dist_tag ? dist_tag : '' %>
```

**Comment by pradeepchhetri** [comment]
@schisamo Updated as per your comments. Can you please review it.

**Comment by schisamo** [comment]
@pradeepchhetri Almost there. We need unit test coverage for the contents of the generated SPEC file. That would go somewhere in here:
https://github.com/chef/omnibus/blob/49f6f2197fc86896aacf6fdc542e964e5547d400/spec/unit/packagers/rpm_spec.rb#L130-L257

**Comment by pradeepchhetri** [comment]
@schisamo Added a unit test for checking the spec file. Sorry for missing so many unit test cases.

**Comment by sersut** [comment]
:+1:

**Comment by sersut** [comment]
Thanks @pradeepchhetri. Merging this and will include this in the release of omnibus that will be coming soon.

**Comment by pradeepchhetri** [comment]
Thank you very much @schisamo @sersut

**Pull request merged**